typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -184,29 +287,296 @@ export class EvmHookModule extends HyperlaneModule<
+        ).join(', ')}...`,
+        chainId: this.domainId,
+        to: gasOracle,
+        data: StorageGasOracle__factory.createInterface().encodeFunctionData(
+          'setRemoteGasDataConfigs((uint32,uint128,uint128)[])',
+          [configsToSet],
+        ),
+      },
+    ];
+  }
+
+  protected async updateProtocolFeeHook({
+    currentConfig,
+    targetConfig,
+  }: {
+    currentConfig: ProtocolFeeHookConfig;
+    targetConfig: ProtocolFeeHookConfig;
+  }): Promise<AnnotatedEV5Transaction[]> {
+    const updateTxs = [];
+    const protocolFeeInterface = ProtocolFee__factory.createInterface();
+
+    // if maxProtocolFee has changed, deploy a new hook
+    if (currentConfig.maxProtocolFee !== targetConfig.maxProtocolFee) {
+      const hook = await this.deployProtocolFeeHook({ config: targetConfig });
+      this.args.addresses.deployedHook = hook.address;
+      return [];
+    }
+
+    // Update protocol fee if changed
+    if (currentConfig.protocolFee !== targetConfig.protocolFee) {
+      updateTxs.push({
+        annotation: `Updating protocol fee from ${currentConfig.protocolFee} to ${targetConfig.protocolFee}`,
+        chainId: this.domainId,
+        to: this.args.addresses.deployedHook,
+        data: protocolFeeInterface.encodeFunctionData(
+          'setProtocolFee(uint256)',
+          [targetConfig.protocolFee],
+        ),
+      });
+    }
+
+    // Update beneficiary if changed
+    if (currentConfig.beneficiary !== targetConfig.beneficiary) {
+      updateTxs.push({
+        annotation: `Updating beneficiary from ${currentConfig.beneficiary} to ${targetConfig.beneficiary}`,
+        chainId: this.domainId,
+        to: this.args.addresses.deployedHook,
+        data: protocolFeeInterface.encodeFunctionData(
+          'setBeneficiary(address)',
+          [targetConfig.beneficiary],
+        ),
+      });
+    }
+
+    // Return the transactions to update the protocol fee hook
+    return updateTxs;
+  }
+
   // Updates a routing hook
   protected async updateRoutingHook({
-    current,
-    target,
+    currentConfig,
+    targetConfig,
   }: {
-    current: DomainRoutingHookConfig | FallbackRoutingHookConfig;
-    target: DomainRoutingHookConfig | FallbackRoutingHookConfig;
+    currentConfig: DomainRoutingHookConfig | FallbackRoutingHookConfig;
+    targetConfig: DomainRoutingHookConfig | FallbackRoutingHookConfig;
   }): Promise<AnnotatedEV5Transaction[]> {
     // Deploy a new fallback hook if the fallback config has changed
     if (
-      target.type === HookType.FALLBACK_ROUTING &&
+      targetConfig.type === HookType.FALLBACK_ROUTING &&
       !configDeepEquals(
-        target.fallback,
-        (current as FallbackRoutingHookConfig).fallback,
+        targetConfig.fallback,
+        (currentConfig as FallbackRoutingHookConfig).fallback,
       )
     ) {
-      const hook = await this.deploy({ config: target });
+      const hook = await this.deploy({ config: targetConfig });
```

Comment:
iiuc, `deploy()` will transfer ownership to the `config.owner`. This is fine if the owner and the subsequent signer of this updateTx are the same. 

But i believe it will be an issue if `config.owner` is set to an address that cannot sign txs, such as SAFE, address(0), etc.

example flow for illustration:
- CLI calls this fn with `signer`
- fn deploys, and then transfers to `config.owner`
- CLI attempts to `sendTransaction(updateTxs)` using `signer`